cmp/form/NumberField.js

```diff
@@ -24,22 +26,31 @@ export class NumberField extends HoistField {
         /** minimum value */
         min: PT.number,
         /** maximum value */
-        max: PT.number
+        max: PT.number,
+        /** Number of decimal places to allow on field's value, defaults to 4*/
+        precision: PT.number,
+        /** Set to true to automatically fill in zeros in accord with precision */
+        zeroPad: PT.bool,
+        /** Constrain input to numeric characters, should be set to false for advanced input evaluation */
+        allowNumericCharactersOnly: PT.bool,
+        /** Whether to display large values with commas */
+        displayWithDelimiters: PT.bool
```

Comment:
Not super sure about the defaults I chose for these new props. I have gonna accustom to the admin use case and am not really sure about what the most commonly desired behavior will be. 

Pretty sure zeroPad should be false.
Precision is defaulted to 4 as that is what we have done for client apps in the past.
As for the commas, and shorthand eval really not sure.